In [25]:
# NAME : ABUSOMWAN AMADEVBORHO
# ASSIGNMENT: PROGRAMMING FOR DATA ANALYTICS PROJECT 1
#School; Technological University of the Shannon

TASK 3

Developed Plans For Hypothesis Testing

H0: The availability of bikes and bike stand in dublin does not depend on time. H1: The availability of bikes and bike stand in dublin depends on time


DEVELOPED QUESTIONS:

1) Do stations with payment terminal increase the number of available bikes?

2)what is the total available bike stand, bike stand and available bikes that has a payment terminal in a working day in contrast to a bank holiday?

3)What is the average values for the dublin bike that has transaction terminal and are also opened on a working/bank holiday period?

4)What address has the highest number of available bikes on monday(bank holiday) and wednesday(working)?

5)what is the average availible bikes for a working day compared to a bank holiday?

6)what is the number of available bikes in Avondale Road for a working day and a bank holiday?

7) What is the maximum difference between available bike stand and bike stand(spread) for banking holiday and working day

8)summary statistics of the available bike stand percentage

To answer these set of related questions.Firstly, I imported the necessary relavant libraries.Futhermore, I call out
our dataset which is the real time bike data later stored as a csv file


In [2]:
import pandas as pd
import numpy as np
from pandas import DataFrame
import warnings
warnings.filterwarnings("ignore")
import matplotlib.pyplot as plt
%matplotlib inline

In [ ]:
pd.read_csv("dublin_bikes.csv")

In [ ]:
pd.read_csv("No_Holiday_bikes.csv")

In [7]:
pd.read_csv("dublin.csv")

Unnamed: 0  NUMBER CONTRACT NAME                    NAME  \
0             0      42        dublin        SMITHFIELD NORTH   
1             1      30        dublin    PARNELL SQUARE NORTH   
2             2      54        dublin          CLONMEL STREET   
3             3     108        dublin           AVONDALE ROAD   
4             4      56        dublin      MOUNT STREET LOWER   
..          ...     ...           ...                     ...   
106         106      39        dublin          WILTON TERRACE   
107         107      83        dublin              EMMET ROAD   
108         108      92        dublin  HEUSTON BRIDGE (NORTH)   
109         109      21        dublin   LEINSTER STREET SOUTH   
110         110      88        dublin         BLACKHALL PLACE   

                    ADDRESS                              POSITION  BANKING  \
0          Smithfield North  {'lat': 53.349562, 'lng': -6.278198}    False   
1      Parnell Square North  {'lat': 53.353462, 'lng': -6.265305}    False   
2            Clonmel Street   {'lat': 53.336021, 'lng': -6.26298}    False   
3             Avondale Road  {'lat': 53.359405, 'lng': -6.276142}    False   
4        Mount Street Lower    {'lat': 53.33796, 'lng': -6.24153}    False   
..                      ...                                   ...      ...   
106          Wilton Terrace  {'lat': 53.332383, 'lng': -6.252717}    False   
107              Emmet Road  {'lat': 53.340714, 'lng': -6.308191}    False   
108  Heuston Bridge (North)  {'lat': 53.347802, 'lng': -6.292432}    False   
109   Leinster Street South   {'lat': 53.34218, 'lng': -6.254485}    False   
110         Blackhall Place    {'lat': 53.3488, 'lng': -6.281637}    False   

     BONUS  BIKE STANDS  AVAILABLE BIKE STANDS STATUS    LAST UPDATE  \
0    False           30                     18   OPEN  1641392086000   
1    False           20                      6   OPEN  1641392209000   
2    False           33                     24   OPEN  1641392166000   
3    False           35                     13   OPEN  1641392209000   
4    False           40                     23   OPEN  1641392174000   
..     ...          ...                    ...    ...            ...   
106  False           20                     13   OPEN  1641391893000   
107  False           40                     23   OPEN  1641392059000   
108  False           40                     27   OPEN  1641392001000   
109  False           30                     14   OPEN  1641392287000   
110  False           30                     22   OPEN  1641392226000   

     AVAILABLE BIKES                DATES  
0                 12  2022-01-05 14:14:46  
1                 14  2022-01-05 14:16:49  
2                  9  2022-01-05 14:16:06  
3                 22  2022-01-05 14:16:49  
4                 17  2022-01-05 14:16:14  
..               ...                  ...  
106                7  2022-01-05 14:11:33  
107               17  2022-01-05 14:14:19  
108               13  2022-01-05 14:13:21  
109               16  2022-01-05 14:18:07  
110                8  2022-01-05 14:17:06  

[111 rows x 14 columns]

In [16]:
DUBLIN_BIKES=pd.read_csv("dublin_bikes.csv")
#DUBLIN_BIKES.head()
#DUBLIN_BIKES.tail()
#DUBLIN_BIKES.columns

WORKING_DAY=pd.read_csv("No_Holiday_bikes.csv")

DUBLIN_BIKES_2=pd.read_csv("dublin.csv")


This API provides real time information on the Dublin bikes.This times series data has a total of 111 rows and 14 columns and it shows the real time dublin bike usage over a period of time.Nevertheless,I noticeed the table has some irrelavant column names in the DataFrame such as "Unnamed" and "Name(which contains same variables as address)".Therefore, this data would undergo cleaning so as to work with the relavant DataFrame for the analysis.

In [17]:
DUBLIN_BIKES_REALTIME=pd.DataFrame(DUBLIN_BIKES,columns=["NUMBER","CONTRACT NAME", "ADDRESS","POSITION",
                    "BANKING","BONUS",'BIKE STANDS',
                   'AVAILABLE BIKE STANDS','STATUS','LAST UPDATE',"AVAILABLE BIKES",
                   "DATES"])
#DUBLIN_BIKES_REALTIME
#DUBLIN_BIKES_REALTIME.head()

DUBLIN_BIKE_STATIC=DUBLIN_BIKES_2.rename(columns= {"Number":"NUMBER", "Address":"ADDRESS","Latitude":"LATITUDE","Longitude":"LONGITUDE"})

DUBLINBIKE_STATIC=pd.DataFrame(DUBLIN_BIKE_STATIC,columns=["NUMBER","ADDRESS","LATITUDE","LONGITUDE"])
#DUBLINBIKE_STATIC

Dublin Bikes are in two kinds which is the real time and static data.Static data provides stable information
like station position,longitude and latitude etc.Fotunately, Static data can be downloaded manually in CSV file format or 
accessedthrough the API. Dynamic data which is our real time data are refreshed every minute and can be accessed only through 
the API(Dublinbikes DCC - Dublinbikes 2021/04/01-2021/07/01 - data.gov.ie, 2022). 
Futhermore,After cleaning, we can use the dataset to get answers for the questions stated in the hypothesis. 
H0: The availability of bikes in dublin does not depend on location. H1: The availability of bikes in dublin depends on location.
To get access to the locations using latitude and longitude we therefore augment the realtime data with a static data so as to
get additional columns for the latitude and longitude which was scrapped from the web. 

The next step, having the static data we augment with the Dublin bike data.This can be merged with a "outer join" 

In [19]:
BIKE_DATA=pd.merge(DUBLIN_BIKES_REALTIME,DUBLIN_BIKE_STATIC, how="outer")
BIKE_DATA

Therefore, with the updated bike data, we now have a combination of both the static and dynamic(real time bike) data.Therefore,
there is an increase in the size of the dataset which we would be working with in the course of this project.

In [20]:
BANK_HOLIDAY=pd.DataFrame(BIKE_DATA,columns=["NUMBER","CONTRACT NAME", "ADDRESS","POSITION",
                    "BANKING","BONUS",'BIKE STANDS',
                   'AVAILABLE BIKE STANDS','STATUS','LAST UPDATE',"AVAILABLE BIKES",
                   "DATES","LATITUDE","LONGITUDE"])
BANK_HOLIDAY

#BANK_HOLIDAY.info()




NUMBER CONTRACT NAME                       ADDRESS  \
0        42        dublin              Smithfield North   
1        30        dublin          Parnell Square North   
2        54        dublin                Clonmel Street   
3       108        dublin                 Avondale Road   
4        56        dublin            Mount Street Lower   
..      ...           ...                           ...   
107      83        dublin                    Emmet Road   
108      92        dublin        Heuston Bridge (North)   
109      21        dublin         Leinster Street South   
110      88        dublin               Blackhall Place   
111      81           NaN  St. James Hospital (Central)   

                                 POSITION BANKING  BONUS  BIKE STANDS  \
0    {'lat': 53.349562, 'lng': -6.278198}   False  False         30.0   
1    {'lat': 53.353462, 'lng': -6.265305}   False  False         20.0   
2     {'lat': 53.336021, 'lng': -6.26298}   False  False         33.0   
3    {'lat': 53.359405, 'lng': -6.276142}   False  False         35.0   
4      {'lat': 53.33796, 'lng': -6.24153}   False  False         40.0   
..                                    ...     ...    ...          ...   
107  {'lat': 53.340714, 'lng': -6.308191}   False  False         40.0   
108  {'lat': 53.347802, 'lng': -6.292432}   False  False         40.0   
109   {'lat': 53.34218, 'lng': -6.254485}   False  False         30.0   
110    {'lat': 53.3488, 'lng': -6.281637}   False  False         30.0   
111                                   NaN     NaN    NaN          NaN   

     AVAILABLE BIKE STANDS STATUS   LAST UPDATE  AVAILABLE BIKES  \
0                     26.0   OPEN  1.641203e+12              4.0   
1                     20.0   OPEN  1.641204e+12              0.0   
2                     30.0   OPEN  1.641203e+12              3.0   
3                     15.0   OPEN  1.641203e+12             20.0   
4                     21.0   OPEN  1.641204e+12             18.0   
..                     ...    ...           ...              ...   
107                   18.0   OPEN  1.641204e+12             22.0   
108                   15.0   OPEN  1.641204e+12             25.0   
109                   28.0   OPEN  1.641204e+12              2.0   
110                   12.0   OPEN  1.641204e+12             18.0   
111                    NaN    NaN           NaN              NaN   

                   DATES   LATITUDE  LONGITUDE  
0    2022-01-03 09:50:42  53.349562  -6.278198  
1    2022-01-03 09:57:43  53.353462  -6.265305  
2    2022-01-03 09:49:31  53.336021  -6.262980  
3    2022-01-03 09:51:37  53.359405  -6.276142  
4    2022-01-03 09:55:09  53.337960  -6.241530  
..                   ...        ...        ...  
107  2022-01-03 09:56:33  53.340714  -6.308191  
108  2022-01-03 09:56:08  53.347802  -6.292432  
109  2022-01-03 09:51:47  53.342180  -6.254485  
110  2022-01-03 09:51:40  53.348800  -6.281637  
111                  NaN  53.339983  -6.295594  

[112 rows x 14 columns]

In [143]:
WORKING_DAYY=pd.DataFrame(WORKING_DAY,columns=["NUMBER","CONTRACT NAME", "ADDRESS","POSITION",
                    "BANKING","BONUS",'BIKE STANDS',
                   'AVAILABLE BIKE STANDS','STATUS','LAST UPDATE',"AVAILABLE BIKES",
                   "DATES"])
WORKING_DAYY

NUMBER CONTRACT NAME                 ADDRESS  \
0        42        dublin        Smithfield North   
1        30        dublin    Parnell Square North   
2        54        dublin          Clonmel Street   
3       108        dublin           Avondale Road   
4        56        dublin      Mount Street Lower   
..      ...           ...                     ...   
106      39        dublin          Wilton Terrace   
107      83        dublin              Emmet Road   
108      92        dublin  Heuston Bridge (North)   
109      21        dublin   Leinster Street South   
110      88        dublin         Blackhall Place   

                                 POSITION  BANKING  BONUS  BIKE STANDS  \
0    {'lat': 53.349562, 'lng': -6.278198}    False  False           30   
1    {'lat': 53.353462, 'lng': -6.265305}    False  False           20   
2     {'lat': 53.336021, 'lng': -6.26298}    False  False           33   
3    {'lat': 53.359405, 'lng': -6.276142}    False  False           35   
4      {'lat': 53.33796, 'lng': -6.24153}    False  False           40   
..                                    ...      ...    ...          ...   
106  {'lat': 53.332383, 'lng': -6.252717}    False  False           20   
107  {'lat': 53.340714, 'lng': -6.308191}    False  False           40   
108  {'lat': 53.347802, 'lng': -6.292432}    False  False           40   
109   {'lat': 53.34218, 'lng': -6.254485}    False  False           30   
110    {'lat': 53.3488, 'lng': -6.281637}    False  False           30   

     AVAILABLE BIKE STANDS STATUS    LAST UPDATE  AVAILABLE BIKES  \
0                       18   OPEN  1641392086000               12   
1                        6   OPEN  1641392209000               14   
2                       24   OPEN  1641392166000                9   
3                       13   OPEN  1641392209000               22   
4                       23   OPEN  1641392174000               17   
..                     ...    ...            ...              ...   
106                     13   OPEN  1641391893000                7   
107                     23   OPEN  1641392059000               17   
108                     27   OPEN  1641392001000               13   
109                     14   OPEN  1641392287000               16   
110                     22   OPEN  1641392226000                8   

                   DATES  
0    2022-01-05 14:14:46  
1    2022-01-05 14:16:49  
2    2022-01-05 14:16:06  
3    2022-01-05 14:16:49  
4    2022-01-05 14:16:14  
..                   ...  
106  2022-01-05 14:11:33  
107  2022-01-05 14:14:19  
108  2022-01-05 14:13:21  
109  2022-01-05 14:18:07  
110  2022-01-05 14:17:06  

[111 rows x 12 columns]

VISUALIZATION OF THE VARIOUS LOCATIONS IN DUBLIN HAVING AVAILABLE BIKES 

In [62]:
#conda install -c plotly plotly=5.5.0

In [63]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [117]:
fig = px.scatter_mapbox(BANK_HOLIDAY, lat="LATITUDE", lon="LONGITUDE", color="AVAILABLE BIKES",mapbox_style="open-street-map", zoom = 17, color_continuous_scale = 'Reds',range_color = [5.0,10.0],animation_frame = 'DATES')

fig.show()

In [145]:
BANK_HOLIDAY["AVAILABLE BIKE STANDS %"]= BANK_HOLIDAY["AVAILABLE BIKE STANDS"]/BANK_HOLIDAY["BIKE STANDS"]
BANK_HOLIDAY["SPREAD IN BIKE STANDS"]= BANK_HOLIDAY["BIKE STANDS"]-BANK_HOLIDAY["AVAILABLE BIKE STANDS"]
BANK_HOLIDAY


In [ ]:
WORKING_DAYY["AVAILABLE BIKE STANDS %"]= WORKING_DAYY["AVAILABLE BIKE STANDS"]/WORKING_DAYY["BIKE STANDS"]
WORKING_DAYY["SPREAD IN BIKE STANDS"]= WORKING_DAYY["BIKE STANDS"]-WORKING_DAYY["AVAILABLE BIKE STANDS"]
WORKING_DAYY

In [72]:
BANK_HOLIDAY=BANK_HOLIDAY.rename(columns= {"CONTRACT NAME":"CONTRACT_NAME", 'BIKE STANDS':'BIKE_STANDS','AVAILABLE BIKE STANDS':
                                                  'AVAILABLE_BIKE_STANDS',"AVAILABLE BIKES":"AVAILABLE_BIKES",
                                                  "AVAILABLE BIKE STANDS %":"AVAILABLE_BIKE_STANDS%","SPREAD IN BIKE STANDS":
                                                  "SPREAD_IN_BIKE_STANDS"})


In [146]:
WORKING_DAYY=WORKING_DAYY.rename(columns= {"CONTRACT NAME":"CONTRACT_NAME", 'BIKE STANDS':'BIKE_STANDS','AVAILABLE BIKE STANDS':
                                                  'AVAILABLE_BIKE_STANDS',"AVAILABLE BIKES":"AVAILABLE_BIKES",
                                                  "AVAILABLE BIKE STANDS %":"AVAILABLE_BIKE_STANDS%","SPREAD IN BIKE STANDS":
                                                  "SPREAD_IN_BIKE_STANDS"})


In [179]:
BANK_HOLIDAY.pivot_table(index=["BANKING","ADDRESS"])

#BANK_HOLIDAY.describe()

AVAILABLE_BIKES  \
BANKING ADDRESS                                              
False   Avondale Road                                 20.0   
        Benson Street                                  4.0   
        Blackhall Place                               18.0   
        Blessington Street                            17.0   
        Bolton Street                                  0.0   
...                                                    ...   
True    Convention Centre                             18.0   
        Dame Street                                    8.0   
        Merrion Square West                           17.0   
        Princes Street / O'Connell Street              9.0   
        St. Stephen's Green South                      3.0   

                                           AVAILABLE_BIKE_STANDS  \
BANKING ADDRESS                                                    
False   Avondale Road                                       15.0   
        Benson Street                                       36.0   
        Blackhall Place                                     12.0   
        Blessington Street                                   3.0   
        Bolton Street                                       20.0   
...                                                          ...   
True    Convention Centre                                   22.0   
        Dame Street                                          8.0   
        Merrion Square West                                  3.0   
        Princes Street / O'Connell Street                   14.0   
        St. Stephen's Green South                           27.0   

                                           AVAILABLE_BIKE_STANDS%  \
BANKING ADDRESS                                                     
False   Avondale Road                                    0.428571   
        Benson Street                                    0.900000   
        Blackhall Place                                  0.400000   
        Blessington Street                               0.150000   
        Bolton Street                                    1.000000   
...                                                           ...   
True    Convention Centre                                0.550000   
        Dame Street                                      0.500000   
        Merrion Square West                              0.150000   
        Princes Street / O'Connell Street                0.608696   
        St. Stephen's Green South                        0.900000   

                                           BIKE_STANDS   LAST UPDATE  \
BANKING ADDRESS                                                        
False   Avondale Road                             35.0  1.641203e+12   
        Benson Street                             40.0  1.641204e+12   
        Blackhall Place                           30.0  1.641204e+12   
        Blessington Street                        20.0  1.641204e+12   
        Bolton Street                             20.0  1.641203e+12   
...                                                ...           ...   
True    Convention Centre                         40.0  1.641204e+12   
        Dame Street                               16.0  1.641203e+12   
        Merrion Square West                       20.0  1.641204e+12   
        Princes Street / O'Connell Street         23.0  1.641204e+12   
        St. Stephen's Green South                 30.0  1.641204e+12   

                                            LATITUDE  LONGITUDE  NUMBER  \
BANKING ADDRESS                                                           
False   Avondale Road                      53.359405  -6.276142     108   
        Benson Street                      53.344153  -6.233451      90   
        Blackhall Place                    53.348800  -6.281637      88   
        Blessington Street                 53.356769  -6.268140       2   
        Bolton Street                      53.351182  -6.269859   

In [147]:
#WORKING_DAYY.pivot_table(index=["BANKING","ADDRESS"])

WORKING_DAYY.describe()

NUMBER  BIKE_STANDS  AVAILABLE_BIKE_STANDS   LAST UPDATE  \
count  111.000000   111.000000             111.000000  1.110000e+02   
mean    64.351351    31.765766              19.819820  1.641354e+12   
std     54.165009     8.136288               8.480135  3.960444e+08   
min      2.000000     1.000000               0.000000  1.637219e+12   
25%     31.500000    29.000000              13.000000  1.641392e+12   
50%     61.000000    30.000000              22.000000  1.641392e+12   
75%     90.500000    40.000000              26.500000  1.641392e+12   
max    507.000000    40.000000              39.000000  1.641392e+12   

       AVAILABLE_BIKES  AVAILABLE_BIKE_STANDS%  SPREAD_IN_BIKE_STANDS  
count       111.000000              111.000000             111.000000  
mean         11.927928                0.605198              11.945946  
std           5.740940                0.182486               5.725283  
min           1.000000                0.000000               1.000000  
25%           8.000000                0.466667               8.000000  
50%          11.000000                0.625000              11.000000  
75%          15.000000                0.733333              15.000000  
max          32.000000                0.975000              32.000000

1) Do stations with payment terminal increase the number of available bikes ?

In [127]:
BANK_HOLIDAY.groupby("BANKING").AVAILABLE_BIKES.agg(["count", "min","max", "mean"])
BANK_HOLIDAY.groupby("BANKING").AVAILABLE_BIKES.agg(["count", "min","max", "mean"]).plot(kind="bar")

count  min   max       mean
BANKING                             
False      106  0.0  38.0  12.603774
True         5  3.0  18.0  11.000000

In [148]:
WORKING_DAYY.groupby("BANKING").AVAILABLE_BIKES.agg(["count", "min","max", "mean"])

count  min  max       mean
BANKING                            
False      106    1   32  12.075472
True         5    6   14   8.800000

looking at the table we can come to the conclusion that on a Bank holiday or working day, the usage of bikes
with  a banking terminal station tend to have lesser number of available bikes
compared to adrresses that do not have teminal Banking Stations

2)what is the total available bike stand, bike stand and available bikes that has a payment terminal in a working day in contrast
to a bank holiday?

In [133]:
grouped_holiday=BANK_HOLIDAY.groupby("BANKING")

grouped_holiday["AVAILABLE_BIKES", "AVAILABLE_BIKE_STANDS","BIKE_STANDS","SPREAD_IN_BIKE_STANDS"].sum()

AVAILABLE_BIKES  AVAILABLE_BIKE_STANDS  BIKE_STANDS  \
BANKING                                                        
False             1336.0                 2057.0       3397.0   
True                55.0                   74.0        129.0   

         SPREAD_IN_BIKE_STANDS  
BANKING                         
False                   1340.0  
True                      55.0

In [149]:
grouped_working=WORKING_DAYY.groupby("BANKING")

grouped_working["AVAILABLE_BIKES", "AVAILABLE_BIKE_STANDS","BIKE_STANDS","SPREAD_IN_BIKE_STANDS"].sum()

AVAILABLE_BIKES  AVAILABLE_BIKE_STANDS  BIKE_STANDS  \
BANKING                                                        
False               1280                   2115         3397   
True                  44                     85          129   

         SPREAD_IN_BIKE_STANDS  
BANKING                         
False                     1282  
True                        44

Here we can see there are more available bike stand on a working day compared to that of a a bank holiday with the 
available bike stand having a total of 85 while that of a bank holiday having a sum of 74 at that particular time. 
Nevertheless,the available bikes are more in a bank holiday compared to a working day.


3)What is the average values for the dublin bike that has transaction terminal and are also opened on a working/bank holiday period?

In [151]:
BANK_HOLIDAY.pivot_table(index=["STATUS","BANKING"])

AVAILABLE_BIKES  AVAILABLE_BIKE_STANDS  \
STATUS BANKING                                           
OPEN   False          12.603774               19.40566   
       True           11.000000               14.80000   

                AVAILABLE_BIKE_STANDS%  BIKE_STANDS   LAST UPDATE   LATITUDE  \
STATUS BANKING                                                                 
OPEN   False                  0.606949     32.04717  1.641166e+12  53.345639   
       True                   0.541739     25.80000  1.641204e+12  53.343544   

                LONGITUDE     NUMBER  SPREAD_IN_BIKE_STANDS  
STATUS BANKING                                               
OPEN   False    -6.264868  65.773585              12.641509  
       True     -6.255923  34.200000              11.000000

In [153]:
WORKING_DAYY.pivot_table(index=["STATUS","BANKING"])

AVAILABLE_BIKES  AVAILABLE_BIKE_STANDS  \
STATUS BANKING                                           
OPEN   False          12.075472               19.95283   
       True            8.800000               17.00000   

                AVAILABLE_BIKE_STANDS%  BIKE_STANDS  BONUS   LAST UPDATE  \
STATUS BANKING                                                             
OPEN   False                  0.603356     32.04717  False  1.641353e+12   
       True                   0.644239     25.80000  False  1.641392e+12   

                   NUMBER  SPREAD_IN_BIKE_STANDS  
STATUS BANKING                                    
OPEN   False    65.773585               12.09434  
       True     34.200000                8.80000

4)What address has the highest number of available bikes on monday(bank holiday) and wednesday(working)?

In [173]:
grouped_holiday1=BANK_HOLIDAY.groupby("CONTRACT_NAME")

grouped_holiday1["ADDRESS","AVAILABLE_BIKES"].max()

ADDRESS  AVAILABLE_BIKES
CONTRACT_NAME                                   
dublin         York Street West             38.0

In [167]:
grouped_working1=WORKING_DAYY.groupby("CONTRACT_NAME")

grouped_working1["ADDRESS","AVAILABLE_BIKES"].max()

ADDRESS  AVAILABLE_BIKES
CONTRACT_NAME                                   
dublin         York Street West               32

5)what is the average availible bikes for a working day?

In [154]:
WORKING_DAYY.AVAILABLE_BIKES.mean()

11.927927927927929

In [155]:
BANK_HOLIDAY.AVAILABLE_BIKES.mean()

12.531531531531531

6)what is the number of available bikes in Avondale Road for a working day and a bank holiday?

In [174]:
WORKING_DAYY[WORKING_DAYY.ADDRESS=="Avondale Road"].AVAILABLE_BIKES

3    22
Name: AVAILABLE_BIKES, dtype: int64

In [92]:
BANK_HOLIDAY[BANK_HOLIDAY.ADDRESS=="Avondale Road"].AVAILABLE_BIKES.mean()

20.0

7) What is the maximum difference between available bike stand and bike stand(spread) for banking holiday and working day

In [182]:
BANK_HOLIDAY["SPREAD_IN_BIKE_STANDS"].max()

38.0

In [181]:
WORKING_DAYY["SPREAD_IN_BIKE_STANDS"].max()

32

8)summary statistics of the available bike stand percentage 

In [201]:
WORKING_DAYY.groupby("AVAILABLE_BIKE_STANDS%").AVAILABLE_BIKES.agg(["count", "max", "mean"])

count  max       mean
AVAILABLE_BIKE_STANDS%                       
0.000000                    1    1   1.000000
0.190476                    1   17  17.000000
0.200000                    1   32  32.000000
0.236842                    1   29  29.000000
0.266667                    1   22  22.000000
0.300000                    2   14  14.000000
0.325000                    1   27  27.000000
0.333333                    1   20  20.000000
0.350000                    1   13  13.000000
0.370370                    1   17  17.000000
0.371429                    1   22  22.000000
0.387097                    1   19  19.000000
0.400000                    5   24  19.200000
0.433333                    1   17  17.000000
0.434783                    1   13  13.000000
0.437500                    1    9   9.000000
0.448276                    1   16  16.000000
0.450000                    3   11  11.000000
0.457143                    1   19  19.000000
0.466667                    3   16  16.000000
0.472222                    1   19  19.000000
0.482759                    1   15  15.000000
0.500000                    6   20  13.500000
0.525000                    1   19  19.000000
0.550000                    1    9   9.000000
0.562500                    1    7   7.000000
0.566667                    2   13  13.000000
0.575000                    3   17  17.000000
0.578947                    1   16  16.000000
0.600000                    5   16  10.400000
0.608696                    1    9   9.000000
0.611111                    1   14  14.000000
0.625000                    3   15  15.000000
0.650000                    4   14  12.250000
0.666667                    3   10   9.333333
0.675000                    5   13  13.000000
0.680000                    1    8   8.000000
0.700000                    4   12  10.500000
0.725000                    5   11  11.000000
0.727273                    1    9   9.000000
0.733333                    6    8   7.666667
0.736842                    1   10  10.000000
0.750000                    3   10   8.333333
0.766667                    5    7   7.000000
0.775000                    3    9   9.000000
0.793103                    1    6   6.000000
0.800000                    3    7   6.333333
0.825000                    1    7   7.000000
0.840000                    1    4   4.000000
0.850000                    1    6   6.000000
0.862069                    1    4   4.000000
0.875000                    1    5   5.000000
0.900000                    1    4   4.000000
0.925000                    2    3   3.000000
0.937500                    1    2   2.000000
0.975000                    1    1   1.000000

In [198]:
grouped40= WORKING_DAYY.groupby(["BIKE_STANDS","AVAILABLE_BIKE_STANDS","ADDRESS"])

In [199]:
grouped40["SPREAD_IN_BIKE_STANDS"].max()

BIKE_STANDS  AVAILABLE_BIKE_STANDS  ADDRESS                                          
1            0                      JCDecaux Ireland, 52 Oriel Street Lower, Dublin 1     1
16           7                      Hardwicke Street                                      9
             9                      Dame Street                                           7
20           6                      Harcourt Terrace                                     14
                                    Parnell Square North                                 14
                                                                                         ..
40           35                     Hanover Quay East                                     5
             36                     Lime Street                                           4
             37                     Heuston Station (Car Park)                            3
                                    The Point                                         

The Dublin bike data which was accessed through an API is a dynamic data or real time data refreshed every minute to give us the bike activities that happens in a particular time. Taking a series of questions and drawing up a hypothesis to test if the availability of bikes and bike stand depends on time having taking data from a working day and also bank holiday. Hence, we can say that there is a significant relationship between time and the availability of bikes and bike stand.Moreover, other factors may also include proximity to bank transaction terminal, address,status etc.


Furthermore, using our data which was categorised into working data and a bank holiday data. This is to see the increase or decrease of bikes and bike stand across various locations at a particular time. Here, we notice that the availability of bikes and bike stand varies with time and occasion using our set of questions developed.



Conclusively, we can say that there is a significant relationship between the availability of bikes and bike stand with time. So with this , we can reject the null hypothesis.



REFERENCING LIST

Data.gov.ie. 2022. Dublinbikes DCC - Dublinbikes 2021/04/01-2021/07/01 - data.gov.ie. [online] Available at: 
<https://data.gov.ie/dataset/dublinbikes-api/resource/1c18f219-3885-403e-aa55-4d4c78ee0204?inner_span=True> 
[Accessed 3 January 2022].
https://towardsdatascience.com/how-i-used-machine-learning-to-improve-my-dublin-bikes-transit-b6bdc7c2b5cb
URL: https://developer.jcdecaux.com/#/opendata/vls?page=getstarted
Developer.jcdecaux.com. 2022. JCDecaux Developer. [online] Available at: <https://developer.jcdecaux.com/> 
[Accessed 2 January 2022].

N:B Remove hashtag to run the operations